Small flake8 issues:
```
partner_communication/models/mail_template.py:12:1: F401 'odoo.fields' imported but unused
partner_communication/wizards/email_template_preview_view.py:19:9: E128 continuation line under-indented for visual indent
partner_communication_revision/wizards/revision_preview.py:60:80: E501 line too long (99 > 79 characters)
```

--- a/partner_communication/models/__init__.py
+++ b/partner_communication/models/__init__.py
@@ -16,5 +16,6 @@ from . import res_partner
 from . import email
 from . import crm_phonecall
 from . import ir_attachment
+from . import mail_template
 from . import communication_dashboard
 from . import report_with_omr

--- a/partner_communication/models/communication_job.py
+++ b/partner_communication/models/communication_job.py
@@ -338,9 +338,10 @@ class CommunicationJob(models.Model):
         self.mapped('attachment_ids').unlink()
         self.set_attachments()
         for job in self:
+            lang = self.env.context.get('lang_preview', job.partner_id.lang)
             if job.email_template_id and job.object_ids:
                 fields = self.env['mail.compose.message'].with_context(
-                    lang=job.partner_id.lang).get_generated_fields(
+                    lang=lang).get_generated_fields(
                     job.email_template_id, [job.id])
                 job.write({
                     'body_html': fields['body_html'],
@@ -354,15 +355,13 @@ class CommunicationJob(models.Model):
     def quick_refresh(self):
         # Only refresh text and subject, all at once
         jobs = self.filtered('email_template_id').filtered('object_ids')
-        langs = set(jobs.mapped('partner_id.lang'))
+        lang = self.env.context.get('lang_preview', jobs.mapped('partner_id.lang'))
         template = jobs.mapped('email_template_id')
-        if len(langs) > 1:
-            raise UserError(_("This is only possible for one lang at time"))
         if len(template) > 1:
             raise UserError(_(
                 "This is only possible for one template at time"))
         values = self.env['mail.compose.message'].with_context(
-            lang=langs.pop()).get_generated_fields(template, jobs.ids)
+            lang=lang).get_generated_fields(template, jobs.ids)
         if not isinstance(values, list):
             values = [values]
         for index in range(0, len(values)):

--- a/partner_communication/models/mail_template.py
+++ b/partner_communication/models/mail_template.py
@@ -1,0 +1,27 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2019 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Joel Vaucher <jvaucher@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import api, models
+
+
+class MailTemplate(models.Model):
+    """change generate_email to can be call with res_ids
+    not only an id but with a record, this is use for the change
+    of email_preview.template res_id selection to reference field"""
+    _inherit = 'mail.template'
+
+    @api.multi
+    def generate_email(self, res_ids, fields=None):
+
+        if isinstance(res_ids, models.BaseModel):
+            res_ids = res_ids.id
+
+        return super(MailTemplate, self).generate_email(res_ids, fields)

--- a/partner_communication/wizards/__init__.py
+++ b/partner_communication/wizards/__init__.py
@@ -12,5 +12,6 @@
 from . import mail_compose_message
 from . import call_wizard
 from . import change_text_wizard
+from . import email_template_preview_view
 from . import pdf_wizard
 from . import generate_communication_wizard

--- a/partner_communication/wizards/email_template_preview_view.py
+++ b/partner_communication/wizards/email_template_preview_view.py
@@ -1,0 +1,43 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2019 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Joel Vaucher <jvaucher@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import models, api, fields
+
+
+class TemplatePreview(models.TransientModel):
+    _inherit = "email_template.preview"
+
+    res_id = fields.Reference(selection=lambda s: s._reference_models(),
+                              default=lambda s: s._default_model())
+
+    @api.model
+    def _reference_models(self):
+        template = self.env['mail.template'].browse(
+            self._context.get('active_id')
+        )
+        domain = [('model', '=', template.model)]
+        models = self.env['ir.model'].search(domain)
+        return [(model.model, model.name) for model in models]
+
+    @api.model
+    def _default_model(self):
+        recs = self._reference_models()
+        model = recs[0][0]
+        specific_domain = {
+            'res.partner': []
+        }
+        domain = specific_domain.get(model) or [('partner_id', '!=', False)]
+        return model + ',' + str(self.env[model].search(domain, limit=1).id)
+
+    @api.onchange('res_id')
+    @api.multi
+    def on_change_res_id(self):
+        return super(TemplatePreview, self).on_change_res_id()

--- a/partner_communication_revision/views/revision_preview_view.xml
+++ b/partner_communication_revision/views/revision_preview_view.xml
@@ -5,12 +5,14 @@
         <field name="arch" type="xml">
             <form>
                 <field name="state" invisible="1"/>
+                <field name="model" invisible="1"/>
                 <h3>Preview of <field name="name" readonly="1" nolabel="1" class="oe_inline"/></h3>
                 <field name="preview_html" widget="html" options='{"safe": True}' attrs="{'readonly': [('state','=','active_revision')]}"/>
                 <hr/>
                 <div states="active_revision">
                     Choose an example of <field name="model_desc" class="oe_inline" readonly="1"/> record:
                     <field name="res_id" class="oe_inline" style="margin-left: 8px;"/>
+                    <hr/>
                 </div>
                 <button name="preview" type="object" string="Refresh preview" class="oe_highlight"/>
                 <group attrs="{'invisible': [('preview_html', '=', False)]}">

--- a/partner_communication_revision/wizards/revision_preview.py
+++ b/partner_communication_revision/wizards/revision_preview.py
@@ -32,9 +32,9 @@ class RevisionPreview(models.TransientModel):
     model = fields.Char(related='revision_id.model', readonly=True)
     model_desc = fields.Char(
         related='revision_id.config_id.model_id.name', readonly=True)
-    res_id = fields.Selection(
-        '_select_res_id', default=lambda s: s._default_res_id()
-    )
+    res_id = fields.Reference(
+        selection='_reference_models',
+        default=lambda s: s._default_model())
     preview_job_id = fields.Many2one(
         'partner.communication.job', readonly=True)
     preview_html = fields.Html(related='preview_job_id.body_html')
@@ -43,26 +43,22 @@ class RevisionPreview(models.TransientModel):
     pdf_page_count = fields.Integer('Number of pages', readonly=True)
 
     @api.model
-    def _select_res_id(self):
+    def _reference_models(self):
         revision_id = self._context.get('revision_id')
         if not revision_id:
             return []
         revision = self.env['partner.communication.revision'].browse(
             int(revision_id))
-        domain = []
-        if revision.model == 'res.partner':
-            domain.append(('lang', '=', revision.lang))
-        else:
-            domain.append(('partner_id.lang', '=', revision.lang))
-        records = self.env[revision.model].search(domain, limit=10,
-                                                  order='id desc')
-        res = [(str(r[0]), r[1]) for r in records.name_get()]
-        return res
+        domain = [('model', '=', revision.model)]
+        models = self.env['ir.model'].search(domain)
+        return [(model.model, model.name) for model in models]
 
     @api.model
-    def _default_res_id(self):
-        recs = self._select_res_id()
-        return recs and recs[0][0]
+    def _default_model(self):
+        recs = self._reference_models()
+        model = recs[0][0]
+        return model + ',' + str(self.env[model].search(
+            [('partner_id', '!=', False)], limit=1).id)
 
     @api.multi
     def unlink(self):
@@ -75,18 +71,20 @@ class RevisionPreview(models.TransientModel):
         if not self.revision_id or not self.res_id:
             return self._reload()
         record = self.env[self.model].browse(int(self.res_id))
-        partner_id = record.id if self.model == 'res.partner' else \
+        partner_id = \
+            record.id if self.model == 'res.partner' else \
             record.partner_id.id
         config = self.revision_id.config_id
         job_vals = {
             'config_id': config.id,
-            'object_ids': self.res_id,
+            'object_ids': self.res_id.id,
             'partner_id': partner_id,
             'state': 'done',  # Prevents sending test jobs,
             'send_mode': 'digital',  # Prevents pdf generation
             'body_html': self._context.get('working_text'),  # Custom text
             'subject': self._context.get('working_subject'),
             'auto_send': False,  # Prevents automatic sending
+            'lang': self.revision_id.lang,
         }
         if not self.preview_job_id:
             # Avoid creating attachments for the communication
@@ -94,13 +92,13 @@ class RevisionPreview(models.TransientModel):
             if attachments_function:
                 config.write({'attachments_function': False})
             # Create a test communication_job record
-            self.preview_job_id = self.preview_job_id.with_context(
-                lang=self.revision_id.lang).create(job_vals)
+            self.preview_job_id = self.preview_job_id.create(job_vals)
             if attachments_function:
                 config.write({'attachments_function': attachments_function})
         elif self.state == 'active_revision':
             self.preview_job_id.write(job_vals)
-            self.preview_job_id.quick_refresh()
+            self.preview_job_id.with_context(
+                {'lang_preview': job_vals['lang']}).quick_refresh()
 
         if config.report_id:
             # Create a PDF preview
@@ -129,3 +127,8 @@ class RevisionPreview(models.TransientModel):
             'type': 'ir.actions.act_window',
             'target': 'new',
         }
+
+    @api.onchange('res_id')
+    def _res_id_change(self):
+        if self.res_id:
+            self._reload()

